bridges/otellogr/logsink_test.go

```diff
@@ -96,3 +112,165 @@ func TestNewLogSink(t *testing.T) {
+	Severity   log.Severity
+	Attributes []log.KeyValue
+}
+
+func TestLogSink(t *testing.T) {
+	now := time.Now()
+
+	for _, tt := range []struct {
+		name            string
+		f               func(*logr.Logger)
+		wantLoggerCount int
+		wantRecords     []wantRecord
+	}{
+		{
+			name: "info",
+			f: func(l *logr.Logger) {
+				l.Info("info message")
+			},
+			wantRecords: []wantRecord{
+				{
+					Body:     log.StringValue("info message"),
+					Severity: log.SeverityInfo,
+				},
+			},
+		},
+		{
+			name: "info_multi_attrs",
+			f: func(l *logr.Logger) {
+				l.Info("msg",
+					"struct", struct{ data int64 }{data: 1},
+					"bool", true,
+					"duration", time.Minute,
+					"float64", 3.14159,
+					"int64", -2,
+					"string", "str",
+					"time", now,
+					"uint64", uint64(3),
+				)
+			},
+			wantRecords: []wantRecord{
+				{
+					Body:     log.StringValue("msg"),
+					Severity: log.SeverityInfo,
+					Attributes: []log.KeyValue{
+						log.String("struct", "{data:1}"),
+						log.Bool("bool", true),
+						log.Int64("duration", 60_000_000_000),
+						log.Float64("float64", 3.14159),
+						log.Int64("int64", -2),
+						log.String("string", "str"),
+						log.Int64("time", now.UnixNano()),
+						log.Int64("uint64", 3),
+					},
+				},
+			},
+		},
+		{
+			name: "info_with_name",
+			f: func(l *logr.Logger) {
+				l.WithName("test").Info("info message with name")
+			},
+			wantRecords: []wantRecord{
+				{
+					Body:     log.StringValue("info message with name"),
+					Severity: log.SeverityInfo,
+				},
+			},
+		},
+		{
+			name: "info_with_name_nested",
+			f: func(l *logr.Logger) {
+				l.WithName("test").WithName("test").Info("info message with name")
+			},
+			wantRecords: []wantRecord{
+				{
+					Body:     log.StringValue("info message with name"),
+					Severity: log.SeverityInfo,
+				},
+			},
+		},
+		{
+			name: "info_with_attrs",
+			f: func(l *logr.Logger) {
+				l.WithValues("key", "value").Info("info message with attrs")
+			},
+			wantRecords: []wantRecord{
+				{
+					Body:     log.StringValue("info message with attrs"),
+					Severity: log.SeverityInfo,
+					Attributes: []log.KeyValue{
+						log.String("key", "value"),
+					},
+				},
+			},
+		},
+		{
+			name: "info_with_attrs_nested",
+			f: func(l *logr.Logger) {
+				l.WithValues("key1", "value1").Info("info message with attrs", "key2", "value2")
+			},
+			wantRecords: []wantRecord{
+				{
+					Body:     log.StringValue("info message with attrs"),
+					Severity: log.SeverityInfo,
+					Attributes: []log.KeyValue{
+						log.String("key1", "value1"),
+						log.String("key2", "value2"),
+					},
+				},
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			rec := logtest.NewRecorder()
+			ls := NewLogSink("name", WithLoggerProvider(rec))
+			l := logr.New(ls)
+			tt.f(&l)
+
+			last := len(rec.Result()) - 1
+
+			assert.Len(t, rec.Result()[last].Records, len(tt.wantRecords))
+			for i, record := range rec.Result()[last].Records {
```

Comment:
We should also assert the `Name`, `Version`, and `SchemeURL` of [ScopeRecords](https://pkg.go.dev/go.opentelemetry.io/otel/log/logtest#ScopeRecords)